src/lib/dialog/dialog-ref.ts

```diff
@@ -72,14 +73,15 @@ export class MatDialogRef<T, R = any> {
 
     _overlayRef.keydownEvents()
       .pipe(filter(event => event.keyCode === ESCAPE && !this.disableClose))
-      .subscribe(() => this.close());
+      .subscribe(() => this.close(undefined, 'keyboard'));
   }
 
   /**
    * Close the dialog.
    * @param dialogResult Optional result to return to the dialog opener.
+   * @param closeFocusOrigin Focus origin that describes how the dialog got closed
```

Comment:
I'm not sure whether this description is super obvious about what it does. Maybe something along the lines of "Type of interaction that closed the dialog"? Also I think the param can be renamed to `focusOrigin` since it's already part of the `close` method.